pkg/prometheus/operator.go

```diff
@@ -902,11 +902,16 @@ func (c *Operator) loadBasicAuthSecrets(mons map[string]*v1alpha1.ServiceMonitor
 					}
 				}
 
-				secrets[fmt.Sprintf("%s/%s/%d", mon.Namespace, mon.Name, i)] =
-					BasicAuthCredentials{
-						username: username,
-						password: password,
-					}
+				if username == "" || password == "" {
```

Comment:
Yes I think that makes more sense. As that sounds like a configuration issue from the administrator using the Prometheus Operator.